java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -324,6 +326,19 @@ private static void cleanupUnfeasibleBreakPaths(List<Block> blocks) {
     }
   }
 
+  private static boolean isLoop(Block successor) {
+    return successor.terminator != null
+      && successor.terminator.is(Tree.Kind.WHILE_STATEMENT, Tree.Kind.DO_STATEMENT, Tree.Kind.FOR_STATEMENT, Tree.Kind.FOR_EACH_STATEMENT);
+  }
+
+  @CheckForNull
+  private static Block getAfterLoopBlock(Block loop) {
+    if (loop.falseBlock != null) {
+      return loop.falseBlock;
+    }
+    return loop.successorWithoutJump; // Because 'for' statements without condition are unconditional jumps block
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move this trailing comment on the previous empty line. ([squid:TrailingCommentCheck](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ATrailingCommentCheck))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=1984&issues=AWKvAdDT_tOugibaIUsv&open=AWKvAdDT_tOugibaIUsv)

[ISSUE_KEY]: AWKvAdDT_tOugibaIUsv